nixos/modules/config/nix.nix

```diff
@@ -386,5 +387,12 @@ in
         )
       );
     };
+
+    # Enable git if flake support is enabled
+    programs.git = mkIf (builtins.match ".*flakes.*" cfg.extraOptions != null) ({
+      enable = mkOverride 1400 true; # mkOptionDefault is 1500, mkDefault is 1000
```

Comment:
Mind elaborating on the reasoning for why `enable` has such an arbitrary priority? Is `mkDefault` not enough?

Reply:
I also don't quite understand why you're setting `package` here at all. Why not leave that to whatever people set via their own `programs.git.package`?

Reply:
The option default value is `1500`. We want to be able to override that. However, some users may set `programs.git.enable = mkDefault <somevalue>`, which should cause the `git` package to be used instead of `gitMinimal`.

Reply:
Right but I'm asking why you feel that this change should also change the default git provided.

Reply:
I'm sure it's because of closure size difference, 300MB vs 50MB, see https://github.com/NixOS/nixpkgs/issues/325924#issuecomment-2218781123
@Frontear

Reply:
It is indeed because of the closure size difference